core/src/main/java/io/grpc/internal/ServerCallImpl.java

```diff
@@ -177,7 +177,11 @@ public void setCompression(String compressorName) {
     checkState(!sendHeadersCalled, "sendHeaders has been called");
 
     compressor = compressorRegistry.lookupCompressor(compressorName);
-    checkArgument(compressor != null, "Unable to find compressor by name %s", compressorName);
+    if (compressor == null) {
+      throw Status.INTERNAL
```

Comment:
This does not seem appropriate. This seems to abuse Status to represent something other than an RPC failure. When this exception is thrown the RPC is still running; the RPC has not been cancelled. I don't think we use Status in this way any where else, and it seems it would be a good idea not to start.

As a side note, it throwing `IllegalStateException` is in the Javadocs in `ServerCall`. If it were changed here, those docs would need to be updated as well.